app/components/Swapbox.tsx

```diff
@@ -272,6 +270,17 @@ export const Swapbox = ({ fixedProductMarketMaker }: SwapboxProps) => {
     setTokenAmountOut(undefined);
   };
 
+  const isEnoughBalance =
+    +tokenAmountIn <= +formatEther(currentState.balance ?? BigInt(0));
+  const isButtonDisabled = !tokenAmountIn || !isEnoughBalance;
+
+  const getButtonLabel = () => {
+    if (!tokenAmountIn) return 'Enter amount';
+    else if (!isEnoughBalance)
+      return `Insufficient ${currentState.inToken.symbol ?? 'pool tokens'} balance`;
+    else return currentState.buttonText;
```

Comment:
I think it's cleaner like

```suggestion
    if (!tokenAmountIn) return 'Enter amount';
    
    if (!isEnoughBalance)
      return `Insufficient ${currentState.inToken.symbol ?? 'pool tokens'} balance`;
    return currentState.buttonText;
```